pkg/inspector/inspector.go

```diff
@@ -0,0 +1,136 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package inspector
+
+import (
+	"context"
+	"sync"
+
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/conduitio/conduit/pkg/record"
+	"github.com/google/uuid"
+)
+
+// Session wraps a channel of records and provides:
+// 1. a way to send records to it asynchronously
+// 2. a way to know if it's closed or not
+type Session struct {
+	C chan *record.Record
+
+	logger  log.CtxLogger
+	onClose func()
+	closed  bool
+}
+
+func (s *Session) Close() {
+	s.onClose()
+	close(s.C)
+	s.closed = true
+}
+
+func (s *Session) IsClosed() bool {
+	return s.closed
+}
```

Comment:
Yes, although if it's _not_ closed it will block forever. Thankfully we have our own helper for this 🙂 https://github.com/ConduitIO/conduit/blob/e145c81cce73bcb8581d217bd266de225ef2f1a7/pkg/foundation/cchan/chan.go#L41